openhands/llm/llm.py

```diff
@@ -169,13 +174,18 @@ def __init__(
 
         completion_unwrapped = self._completion
 
-        def attempt_on_error(retry_state):
-            """Custom attempt function for litellm completion."""
+        def log_retry_attempt(retry_state):
+            """With before_sleep, this is called before `custom_completion_wait` and
+            ONLY if the retry is triggered by an exception."""
```

Comment:
"ONLY if the retry is triggered by an exception" - isn't the retry always triggered by an exception?

Reply:
The `before_sleep` only fires on exceptions, the former `after` fired for every attempt, even the succesful one.